### PR TITLE
update kind jobs

### PR DESCRIPTION
**Pull request by BenTheElder**
update kubernetes/kubernetes and kubernetes-sigs/kind jobs to use quick build

/cc @amwat @spiffxp

### CODE REVIEW

File: config/jobs/kubernetes/sig-arch/conformance-audit.yaml [hunk 1]
```diff
@@ -34,7 +32,7 @@ periodics:
       - name: GINKGO_TOLERATE_FLAKES
         value: "n"
       - name: BUILD_TYPE
-        value: bazel
+        value: docker
```
Comment (BenTheElder) [inline config/jobs/kubernetes/sig-arch/conformance-audit.yaml:35]: this one is not just dropped because it's using a custom script. cc @hh

**Review by amwat — COMMENTED** [review]
/lgtm

File: config/jobs/kubernetes/sig-arch/conformance-audit.yaml (hunk 1 above)
Comment (hh) [reply config/jobs/kubernetes/sig-arch/conformance-audit.yaml:35]: @zachmandeville can you ensure our jobs are still happy?

Comment (zachmandeville) [reply config/jobs/kubernetes/sig-arch/conformance-audit.yaml:35]: this job is happy! https://k8s-testgrid.appspot.com/sig-arch-conformance#kind-conformance-audit

Comment (BenTheElder) [reply config/jobs/kubernetes/sig-arch/conformance-audit.yaml:35]: and importantly building with make as intended, we are dropping bazel in k/k (notice going out tomorrow)

**Pull request merged**